rskj-core/src/test/java/co/rsk/peg/bitcoin/UtxoUtilsTest.java

```diff
@@ -0,0 +1,176 @@
+package co.rsk.peg.bitcoin;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.config.BridgeMainNetConstants;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.spongycastle.util.encoders.Hex;
+
+class UtxoUtilsTest {
+
+    private final static Coin MAX_BTC = BridgeMainNetConstants.getInstance().getMaxRbtc();
+
+    private static List<Coin> coinListOf(long ... valuesInSatoshis) {
+        return Arrays.stream(valuesInSatoshis)
+            .mapToObj(Coin::valueOf)
+            .collect(Collectors.toList());
+    }
+
+    private static Stream<Arguments> validOutpointValues() {
+        List<Arguments> arguments = new ArrayList<>();
+
+        arguments.add(Arguments.of(Hex.decode("00"), Collections.singletonList(Coin.ZERO)));
+
+        arguments.add(Arguments.of(Hex.decode("01"), Collections.singletonList(Coin.SATOSHI)));
+
+        arguments.add(Arguments.of(Hex.decode("01010101010101010101"), Stream.generate(() -> Coin.SATOSHI).limit(10).collect(Collectors.toList())));
+
+        arguments.add(Arguments.of(Hex.decode("FC"), coinListOf(252)));
+
+        // 252 = FC, 187 = BB, 13_337 = FE9145DC00, 14_435_729 = FEDC4591
+        arguments.add(Arguments.of(Hex.decode("FCFCBBBBBBFD1934FE9145DC00"), coinListOf(252, 252, 187, 187, 187, 13_337, 14_435_729)));
+
+        arguments.add(Arguments.of(Hex.decode("FF0040075AF0750700"), Collections.singletonList(MAX_BTC)));
+
+        arguments.add(Arguments.of(Hex.decode("FF0040075AF0750700FF0040075AF0750700FF0040075AF0750700"), Arrays.asList(MAX_BTC, MAX_BTC, MAX_BTC)));
+
+        arguments.add(Arguments.of(Hex.decode("FF0140075AF0750700"), Collections.singletonList(MAX_BTC.add(Coin.SATOSHI))));
+
+        arguments.add(Arguments.of(Hex.decode("FFFFFFFFFFFFFFFF7F"), coinListOf(Long.MAX_VALUE)));
+
+        final byte[] bigOutpointValueArray = Hex.decode(
+            Stream.iterate("FCFCBBBBBBFD1934FE9145DC00", UnaryOperator.identity()).limit(1000)
+                .collect(Collectors.joining()));
+        List<Coin> bigListOfOutpointValues = Stream.generate(
+                () -> coinListOf(252, 252, 187, 187, 187, 13_337, 14_435_729)).limit(1000)
+            .flatMap(Collection::stream).collect(Collectors.toList());
+        arguments.add(Arguments.of(
+            bigOutpointValueArray,
+            bigListOfOutpointValues
+        ));
```

Comment:
```
infiniteStream = Stream.iterate("FCFCBBBBBBFD1934FE9145DC00", UnaryOperator.identity());
limitedInfiniteStream = infiniteStream.limit(1000);
final byte[] bigOutpointValueArray = Hex.decode(limitedInfiniteStream.collect(Collectors.joining()));
```

please use meaningful name variables when possible

Reply:
sorry to be nit, but theres a bunch of numbers and letters all over the file. It visually helps to have them placed in variables

Reply:
I don't follow. The reason why this has a variable it's to give context. `bigOutpointValueArray` expressed that we are dealing with a big array with many values. 

`infiniteStream = Stream.iterate("FCFCBBBBBBFD1934FE9145DC00", UnaryOperator.identity());` This is not a `iniftiveStream` and to me separate a lambda in this way just a more confusion.

Reply:
actually you are right. I was confused with all the numbers and letters.